File: frontend/src/state/machine_room.rs
/// 机房配置数据模型
#[derive(Clone, Debug, PartialEq)]
pub struct MachineRoomConfig {
    pub id: i32,
    pub room_name: String,
    pub room_code: String,
    pub facility_type: String,
    pub address: String,
    pub provider_id: i32,
    pub room_type: String,
    pub contact_person: String,
    pub contact_phone: String,
    pub floor: Option<String>,
    pub cabinet_count: Option<i32>,
    pub area_size: Option<String>,
    pub remarks: Option<String>,
    pub status: String,
    pub created_at: String,
    pub updated_at: Option<String>,
}

impl MachineRoomConfig {
    pub fn display_name(&self) -> String {
        format!("{} - {}", self.room_name, self.facility_type)
    }
}

File: apps/web/apps/web-antd/src/api/scan/machine-room.ts
import { requestClient } from '#/api/request';

export namespace ScanMachineRoomApi {
  export interface MachineRoom {
    id?: number; room_name: string; room_code: string;
    facility_type: string; address: string; platform_id: number;
    room_type: string; contact_person: string; contact_phone: string;
    floor?: string; cabinet_count?: number; area_size?: string;
    remarks?: string; status: string; created_at?: string; updated_at?: string;
  }
}
export function getMachineRoomList() { return requestClient.get<ScanMachineRoomApi.MachineRoom[]>('/asset-ops/machine-rooms'); }
export function getMachineRoom(id: number) { return requestClient.get<ScanMachineRoomApi.MachineRoom>(`/asset-ops/machine-rooms/${id}`); }
export function createMachineRoom(data: ScanMachineRoomApi.MachineRoom) { return requestClient.post('/asset-ops/machine-rooms', data); }
export function updateMachineRoom(id: number, data: ScanMachineRoomApi.MachineRoom) { return requestClient.put(`/asset-ops/machine-rooms/${id}`, data); }
export function deleteMachineRoom(id: number) { return requestClient.delete(`/asset-ops/machine-rooms/${id}`); }
